x/rewards/keeper/state_flatfee.go

```diff
@@ -44,3 +47,49 @@ func (s FlatFeeState) RemoveFee(contractAddr sdk.AccAddress) {
 	store := prefix.NewStore(s.stateStore, types.FlatFeePrefix)
 	store.Delete(contractAddr.Bytes())
 }
+
+// Import initializes state from the genesis flat fees data.
+func (s FlatFeeState) Import(metadata []types.ContractMetadata, flatFees []types.FlatFee) {
+	contractAddrToMetadata := make(map[string]types.ContractMetadata, 0)
```

Comment:
The validation should be applied in the rewards module `ValidateGenesis` step rather than during the import. ref: https://github.com/archway-network/archway/blob/main/x/rewards/types/genesis.go#L46

Reply:
@fdymylja In this case, should we validate the genesis before calling all import functions in `x/rewards/keeper/genesis.go`'s `Import` method?

Reply:
no, it's not needed, genesis validation is a separate state that happens before the import. ref: https://github.com/archway-network/archway/blob/main/x/rewards/module.go#L60

Reply:
Yeah makes sense. But still I think check is not redundant if requirement is to verify that metadata is present for every addr <> flatfee mapping. Will move this check to https://github.com/archway-network/archway/pull/275/files#diff-805f961f8d4413593130e2fd0cb15f43fa255fbc70caadfb82f33ed50f9da03aR121, let me know if any concern?

Reply:
Updated here b43d7672ae2570811f348383fa9ef96f4f3b3759